tests/js/client/shell/shell-deadlock-trx-hotbackup-cluster.js

```diff
@@ -0,0 +1,129 @@
+      db._drop(cn);
+      c = db._create(cn, {numberOfShards: 1, replicationFactor: 2},
+                     {waitForSyncReplication: true});
+    },
+
+    tearDown: function () {
+      if (c !== null) {
+        c.drop();
+      }
+      c = null;
+      internal.wait(0);
+    },
+
+    testRunAQLWritingTransactionsDuringHotbackup: function () {
+      if (!isCluster) {
+        return true;
+      }
+
+      let start = time();
+      // This will create lots of hotbackups for approximately 60 seconds
+      let res = arango.POST_RAW("/_api/transaction",
+        { action: `function() {
+                    let console = require("console");
+                    let internal = require("internal");
+                    let wait = internal.wait;
+                    let time = internal.time;
+                    let start = time();
+                    let good = 0;
+                    let bad = 0;
+                    while (time() - start < 60) {
+                      try {
+                        internal.createHotbackup({});
+                        console.log("Created a hotbackup!");
+                        good++;
+                      } catch(e) {
+                        console.error("Caught exception when creating a hotbackup!", e);
+                        bad++;
+                      }
+                    }
+                    return {good,bad};
+                  }`,
+          collections: {} }, {"x-arango-async":"store"});
+      assertFalse(res.error, "Could not POST transaction.");
+      assertEqual(202, res.code, "Bad response code.");
+      let jobid = res.headers["x-arango-async-id"];
+
+      // Now we try to write something:
+      for (let i = 0; i < 1000; ++i) {
+        c.insert({Hallo:i});
+      }
+      // And now to execute exclusive write locking transactions:
+      for (let i = 1; i <= 750; ++i) {
+        db._query(`LET m = (FOR d IN ${cn}
+                              SORT d.Hallo DESC
+                              LIMIT 1
+                              RETURN d)
+                   INSERT {Hallo: m[0].Hallo+1} INTO ${cn}
+                   OPTIONS {exclusive: true}
+                   RETURN NEW`).toArray();
+        if (i % 50 === 0) {
+          print("Done", i, "write transactions.");
+        }
+        wait(0.01);
+      }
+      let diff = time() - start;
+      print("Done 750 exclusive transactions in", diff, "seconds!");
+      assertTrue(diff < 60, "750 transactions took too long, probably some deadlock with hotbackups");
+      while (time() - start < 80) {
+        res = arango.PUT(`/_api/job/${jobid}`, {});
+        if (res.code !== 204) {
+          assertEqual(200, res.code, "Response code bad.");
+          assertEqual(0, res.result.bad, "Not all hotbackups went through!");
+          print("Managed to perform", res.result.good, "hotbackups.");
+          break;
+        }
+        wait(1.0);
+        print("Waiting for hotbackups to finish...");
+      }
+    }
```

Comment:
I think we should have an assertion after the `while` loop to check that we didn't use the full 80 seconds.
In addition, we should add a `finally` block to kill/cancel the async job should it still be ongoing.

Reply:
Done.